File: Stack/17608_막대기/17608_막대기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Stack;

public class Main {
	public static void main(String args[]) throws Exception	{
    // set needed variables
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    Stack<Integer> s = new Stack<>();
    int N = Integer.parseInt(br.readLine());
    int input, answer = 0, top;

    // get data
    for (int i = 0; i < N; i++){
      input = Integer.parseInt(br.readLine());
      s.push(input);
    }

    // do process
    top = 0;
    while (!s.empty()){
      int now = s.pop();
      if (now > top){
        top = now;
        answer++;
      }
    }

    // print answer and end program
    System.out.print(answer);
    br.close();
    return;
  }
}

File: Implementation/17608_막대기/17608_막대기.py
# 빠르게 입력 받기
import sys
input = sys.stdin.readline

# N을 입력받으세요
N = int(input())

# stack list를 생성하세요
stack = []

# N개의 입력을 받아 처리하세요
for _ in range(N):
    # 막대의 높이를 받으세요
    bar = int(input())

    # 스택을 활용하세요
    while stack and stack[-1] <= bar:
        stack.pop()
    stack.append(bar)

# 스택에 남아있는 막대의 수를 출력하세요
print(len(stack))